### ISSUE TITLE
Improve error message for "toolset not available"

### ISSUE THREAD

**Issue by TysonAndre** [php/setup-php-sdk#9]
I ran into this since `windows-latest` used to work - I should have just used `runs-on: windows-2019` for my php 7 builds instead.

> Currently, windows-2019 may be used for all PHP versions, although this may change in future releases.

https://github.com/php/setup-php-sdk?tab=readme-ov-file#inputs explains this - perhaps link to that section and https://github.com/php/setup-php-sdk?tab=readme-ov-file#manually-installing-toolsets in the error message itself (I already figured out 

Proposed solution: Attempt to install the toolset version if it doesn't exist (possibly controlled by an `auto_install_dependencies: true` config setting for the job to avoid accidental pollution and retry once

```
Run php/setup-php-sdk@v0.9
Run .....
At D:\a\_actions\php\setup-php-sdk\v0.9\run.ps1:45 char:5
+     throw "toolset not available"
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (toolset not available:String) [], RuntimeException
    + FullyQualifiedErrorId : toolset not available
```

**Comment by TysonAndre** [comment]
https://github.com/php/setup-php-sdk?tab=readme-ov-file#inputs

> Currently, `windows-2019` may be used for all PHP versions, although this may change in future releases.

An unrelated thing I found when investigating this: https://stackoverflow.com/questions/66025220/paired-values-in-github-actions-matrix - using maps for the image version and the php version pairs is another alternative that could be mentioned, especially if multiple architectures or TS/NTS builds are supported - the currently suggested approach in https://github.com/php/setup-php-sdk/?tab=readme-ov-file#inputs is multiple excludes.

```yml
  windows:
    defaults:
      run:
        shell: cmd
    strategy:
      fail-fast: false
      matrix:
          version:
          - { os: windows-2019, php: "7.2" }
          - { os: windows-2019, php: "7.3" }
          - { os: windows-2019, php: "7.4" }
          - { os: windows-2022, php: "8.0" }
          - { os: windows-2022, php: "8.1" }
          - { os: windows-2022, php: "8.2" }
          - { os: windows-2022, php: "8.3" }
          - { os: windows-2022, php: "8.4" }
          arch: [x64]
          ts: [nts, ts]
    # https://github.com/php/setup-php-sdk?tab=readme-ov-file#inputs
    runs-on: ${{matrix.version.os}}
    steps:
      - name: Checkout project
        uses: actions/checkout@v2
      - name: Setup PHP
        id: setup-php
        uses: php/setup-php-sdk@v0.9
        with:
          version: ${{matrix.version.php}}
          arch: ${{matrix.arch}}
          ts: ${{matrix.ts}}

```

**Comment by cmb69** [comment]
> Proposed solution: Attempt to install the toolset version if it doesn't exist (possibly controlled by an `auto_install_dependencies: true` config setting for the job to avoid accidental pollution and retry once

Actually, I don't think this is a good idea, unless this can be cached. https://github.com/php/pecl-database-dbase/actions/runs/10093674036/job/27909899287 reports that the installation takes almost 8 min; this is unbearably slow, in my opinion.

Using a proper matrix might be the better way; see also https://github.com/php/php-windows-builder, which might generally be the favorable solution (although somebody reported recently that it "wouldn't work").

**Comment by TysonAndre** [comment]
The comment in https://github.com/php/setup-php-sdk?tab=readme-ov-file#inputs is also outdated as of php 8.4. php 8.4 results in "toolset not available" on `runs-on: windows-2019`
(Example just to prove the absence of the sdk: https://github.com/TysonAndre/var_representation/actions/runs/10341584115/job/28623549031)

My previous comment had the solution of specifying pairs of os and php version configs that work together: https://github.com/php/setup-php-sdk/issues/9#issuecomment-2282222968

> Currently, windows-2019 may be used for all PHP versions, although this may change in future releases.

should say something along the lines of:

> Currently, windows-2019 may be used for all PHP versions up to PHP 8.3. PHP 8.4 requires a newer image such as windows-2022.

**Comment by cmb69** [comment]
Closed via aeb631f88f71bbf027937e7f7b1553e1d394c99e.
